src/azure-cli-core/azure/cli/core/cli_subprocess.py

```diff
@@ -0,0 +1,67 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import subprocess
+from knack.log import get_logger
+from azure.cli.core.decorators import cli_subprocess_decorator
+
+__all__ = [
+    # recommended invoking function
+    "run",
+    # advanced use cases
+    "CliPopen",
+    # older high-level API, can be replaced by run in many cases, support for existing codes
+    "call",
+    "check_call",
+    "check_output",
+]
+
+logger = get_logger(__name__)
+
+
+@cli_subprocess_decorator
+def run(*popenargs, **kwargs):
+    """Run command with arguments and remove shell=True
+
+    The other arguments are the same as for the subprocess.run.
+    """
+    return subprocess.run(*popenargs, **kwargs)
```

Comment:
When subprocessing `azcopy` and `bicep`, high-level APIs are used.

- `azcopy` (though it is using an [older high-level API](https://docs.python.org/3/library/subprocess.html#older-high-level-api)): https://github.com/Azure/azure-cli/blob/f1d068ece6789d8e77f3a1ee7c140b12982298a8/src/azure-cli/azure/cli/command_modules/storage/azcopy/util.py#L105

- `bicep`: https://github.com/Azure/azure-cli/blob/cff8d9bad3e39aa69ab9ca108693c66088d615bc/src/azure-cli/azure/cli/command_modules/resource/_bicep.py#L327-L331

`azdev` only uses `subprocess.call`: https://github.com/Azure/azure-cli-dev-tools/blob/db1212ac069ce2f4149315b5860c76fc1499b736/azdev/utilities/command.py#L34

Exposing only one function for `subprocess.call` may be sufficient.

Reply:
I think it's ok to expose these commonly used three older high-level APIs cause subprocess itself does this. And it's easier for developers to adjust their current code snippet.